### ISSUE TITLE
Add an integration test for Bisect_ppx, Dune, and Reason

### ISSUE THREAD

**Issue by aantron** [aantron/bisect_ppx#132]
There's no reason (argh) to think that Bisect has a problem with this, but we should at least check it.

**Comment by osener** [comment]
I recently converted a Dune project from OCaml to Reason, and forgot to check if bisect_ppx works with it. But it looks like there are some issues with [`ppx_expect`](https://github.com/janestreet/ppx_expect) when the project is built with `BISECT_ENABLE=YES`:

```
run alias lib/schema/runtest (exit 2)
(cd _build/default/lib/schema && .graphql_schema.inline-tests/run.exe inline-test-runner graphql_schema -source-tree-root ../.. -diff-cmd -)
File "lib/schema/util.re", line 129, characters 2-394: with_scheme threw
(Failure
   "Trying to run an expect test from the wrong file.\
  \n- test declared at lib/schema/util.re:129\
  \n- trying to run it from lib/schema/util.re.ml\
  \n").
  Raised at file "stdlib.ml", line 33, characters 17-33
  Called from file "runtime-lib/runtime.ml", line 327, characters 8-12
  Re-raised at file "runtime-lib/runtime.ml", line 330, characters 6-13
  Called from file "runtime-lib/runtime.ml", line 343, characters 15-52
  Called from file "runtime-lib/runtime.ml", line 430, characters 52-83
```
I'm not sure whether this is something `ppx_expect` or `bisect_ppx` should support. If this doesn't ring a bell I can try to create a small project to reproduce it.

**Comment by aantron** [comment]
I don't see the exact problem immediately, but the error message strongly suggests that it is an interaction between how Dune runs Reason (outputting `util.re.ml`), perhaps how locations are handled by Bisect_ppx, and how ppx_expect is run. This probably needs a repro project for efficient debugging :/

cc @rgrinberg in case this is something obvious.

**Comment by osener** [comment]
@aantron Here it is: https://gist.github.com/osener/bbce07095b9273c1e78f616f31dda100

```bash
$ git clone https://gist.github.com/bbce07095b9273c1e78f616f31dda100.git dune_reason_bisect
$ cd dune_reason_bisect && opam switch create . 4.07.1
$ opam switch create --yes . 4.07.1
$ dune runtest

> 
run alias runtest (exit 2)
(cd _build/default && .dune_reason_bisect.inline-tests/run.exe inline-test-runner dune_reason_bisect -source-tree-root . -diff-cmd -)
File "main.re", line 3, characters 0-81 threw
(Failure
   "Trying to run an expect test from the wrong file.\
  \n- test declared at main.re:3\
  \n- trying to run it from main.re.ml\
  \n").
  Raised at file "stdlib.ml", line 33, characters 17-33
  Called from file "runtime-lib/runtime.ml", line 327, characters 8-12
  Re-raised at file "runtime-lib/runtime.ml", line 330, characters 6-13
  Called from file "runtime-lib/runtime.ml", line 343, characters 15-52
  Called from file "runtime-lib/runtime.ml", line 430, characters 52-83

FAILED 1 / 1 tests
```

**Comment by rgrinberg** [comment]
This seems like a bug in ppx_expect. cc @diml

**Comment by diml** [comment]
This was in fact a ppxlib issue. It was fixed in 0.4.0.
